pkg/cgrouprate/cgrouprate.go

```diff
@@ -117,6 +131,11 @@ func (r *CgroupRate) process(ctx context.Context) {
 	r.log.Infof("Cgroup rate started (%d/%s)",
 		r.opts.Events, time.Duration(r.opts.Interval).String())
 
+	defer func() {
+		// cleanup
+		glSt.handle = nil
+	}()
+
```

Comment:
hum, why is this needed?

Reply:
So the new code has a check that glSt is not nil when we register:

```
	if glSt.handle != nil {
		return nil, fmt.Errorf("cgrouprate: handle is already set, need to cleanup first")
	}
```

This was failing for TestThrottle2 so I added the code above. The idea is that if the goroutine goes away, we can cleanup handle. Alternatively, I could also remove the above check.